functorch/dim/magic_trace.py

```diff
@@ -21,9 +21,10 @@ def magic_trace(output="trace.fxt", magic_trace_cache="/tmp/magic-trace"):
                 magic_trace_cache,
                 "-q",
                 "https://github.com/janestreet/magic-trace/releases/download/v1.0.2/magic-trace",
-            ]
+            ],
+            check=True,
         )
-        subprocess.run(["chmod", "+x", magic_trace_cache])
+        subprocess.run(["chmod", "+x", magic_trace_cache], check=True)
```

Comment:
Most of these that are just running commands should really use subprocess.checkoutput. Please change the calls to do that. Similar actions were taken in the linked PR.